packages/api/src/providers/fitbit.ts

```diff
@@ -444,48 +444,100 @@ export class Fitbit extends Provider implements OAuth2 {
    */
-  async getActiveSubscriptions(
-    accessToken: string
-  ): Promise<FitbitWebhookSubscriptions | undefined> {
-    const url = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions.json`;
-    try {
-      const resp = await axios.get(url, {
-        headers: {
-          Authorization: `Bearer ${accessToken}`,
-        },
+  async getActiveSubscriptions(token: FitbitToken): Promise<FitbitWebhookSubscriptions> {
+    const rejected: { url: string; err: unknown }[] = [];
+    const activeSubscriptions: FitbitWebhookSubscriptions[] = [];
+    await Promise.allSettled(
+      Object.entries(Fitbit.subscriptionTypes).map(async ([scope, collectionType]) => {
+        if (token.scope.includes(scope)) {
+          const url = `${Fitbit.URL}/${Fitbit.API_PATH}/${collectionType}/apiSubscriptions.json`;
+          try {
+            const resp = await axios.get(url, {
+              headers: {
+                Authorization: `Bearer ${token.accessToken}`,
+              },
+            });
+            if (resp.status === status.OK && resp.data.apiSubscriptions.length) {
+              const activeSubscription = fitbitWebhookSubscriptionsSchema.parse(
+                resp.data.apiSubscriptions
+              );
+              activeSubscriptions.push(activeSubscription);
+            }
+          } catch (err) {
+            rejected.push({ url, err });
+          }
+        }
+      })
+    );
+    if (rejected.length > 0) {
+      console.log(`Failed to get active Fitbit WH subscriptions.`);
+      capture.message(`Failed to get active Fitbit WH subscriptions.`, {
+        extra: { context: `fitbit.getActiveSubscriptions`, cause: rejected },
+      });
+      throw new Error("Failed to get active Fitbit WH subscriptions", { cause: rejected });
+    }
+    const activeSubs = activeSubscriptions.flat();
+    return activeSubs;
+  }
+
+  /**
+   * Creates new WH subscriptions based on the user's selected scopes
+   *
+   * @param token: FitbitToken
+   */
+
+  async createSubscriptions(token: FitbitToken): Promise<void> {
+    const rejected: { subscriptionUrl: string; err: unknown }[] = [];
+    await Promise.allSettled(
+      Object.entries(Fitbit.subscriptionTypes).map(async ([key, subscriptionType]) => {
+        if (token.scope.includes(key)) {
+          const subscriptionId = `${token.userId}-${subscriptionType}`;
+          const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
+          try {
+            await this.createSubscription(subscriptionUrl, token.accessToken);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } catch (err: any) {
+            rejected.push({ subscriptionUrl, err: err.cause });
+          }
+        }
+      })
+    );
+
+    if (rejected.length > 0) {
+      capture.message(`Failed to create Fitbit WH subscriptions`, {
+        extra: { context: `Fitbit.createSubscriptions`, cause: rejected },
```

Comment:
Since we're throwing, maybe we don't need to report to Sentry? Otherwise we'd end up with duplicated issues there, this is not good.

nit: we're not logging anything, here or on line 530 - I think we should.